### PR TITLE
Small optimizations

### PR DESCRIPTION
**Pull request by tinwelint**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java [hunk 1]
```diff
@@ -193,10 +195,10 @@ public void init( SPI spi )
 
         this.spi = spi;
         this.relActions = new StandardRelationshipActions( spi::currentStatement, spi::currentTransaction,
-                this::assertTransactionOpen, ( id ) -> new NodeProxy( nodeActions, id ), this );
+                spi::assertInUnterminatedTransaction, ( id ) -> new NodeProxy( nodeActions, id ), this );
         this.nodeActions =
-                new StandardNodeActions( spi::currentStatement, spi::currentTransaction, this::assertTransactionOpen,
-                        relActions, this );
+                new StandardNodeActions( spi::currentStatement, spi::currentTransaction,
+                        spi::assertInUnterminatedTransaction, relActions, this );
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java:201]: I wonder if we can't collect these functions in an interface at this point. Or just hand these guys the SPI reference.

**Comment by MishaDemianenko** [comment]
Should go after GA

File: community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ClassicCoreSPI.java [hunk 2]
```diff
@@ -184,7 +184,6 @@ public KernelTransaction beginTransaction( KernelTransaction.Type type, AccessMo
     @Override
     public KernelTransaction currentTransaction()
     {
-        availability.assertDatabaseAvailable();
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ClassicCoreSPI.java:187]: Please explain the removal of this line.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/ShortArray.java [hunk 3]
```diff
@@ -782,17 +783,15 @@ public static ShortArray typeOf( Object array )
...
-        // bbbb][bbll,llll][yyyy,tttt][kkkk,kkkk][kkkk,kkkk][kkkk,kkkk]
-        bits.getInt( 24 ); // Get rid of key
-        bits.getByte( 4 ); // Get rid of short array type
-        bits.getByte( 4 ); // Get rid of the type
-        int arrayLength = bits.getByte( 6 );
-        int requiredBits = bits.getByte( 6 );
+        int highInt = (int) (firstBlock >>> 32);
+        int arrayLength = highInt & 0b11_1111;
+        highInt >>>= 6;
+        int requiredBits = highInt & 0b11_1111;
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/ShortArray.java:789]: Please add a comment detailing what this is doing, and its approach to doing it.

**Comment by chrisvest** [comment]
@tinwelint I had some comments.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ClassicCoreSPI.java (hunk 2 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ClassicCoreSPI.java:187]: The commit message does

**Comment by tinwelint** [comment]
Closing this since the ShortArray improvement can go to 2.3 instead. Will also add comments.

**Pull request closed without merging**